### ISSUE TITLE
Unfair results due to using MATH as the validation set

### ISSUE THREAD

**Issue by LLMforScience** [TIGER-AI-Lab/CritiqueFineTuning#6]
Based on Table 1, the improvement on MATH is significant, but you use MATH as the validation dataset.

This is unreasonable. A proper validation set should be independent of the test data to ensure a fair assessment.

**Comment by wenhuchen** [comment]
First of all, we used MATH-500, which we didn't report.
Secondly, our SFT baselines are also using MATH-500 for checkpoint selection. It's highly controlled experiments.

**Comment by LLMforScience** [comment]
In the evaluation part, you used the [math-500](https://github.com/TIGER-AI-Lab/CritiqueFineTuning/tree/main/tools/evaluate_math/data/math-500) as evaluation  and report it as final performance. 

This approach is unreasonable in machine learning. A proper validation set should be independent of the test data to ensure a fair assessment. Even though your baselines also used this setup, it still raises concerns about potential overfitting and inflated performance metrics.

**Comment by wenhuchen** [comment]
We report MATH (containing 5000 examples), not MATH-500. They are just in-distribution.
It's just like all the previous datasets being divided into train/val/test, where all of them are in-distribution. 
Furthermore, our training dataset didn't even contain MATH training set. The other methods we compared against all adopt MATH training set during training.

**Comment by LLMforScience** [comment]
The full test set (5K MATH) contains the exact same questions as MATH-500. 

This raises concerns about data leakage, as the evaluation is not truly independent. A fair evaluation should use a distinct test set that does not overlap with the validation data.


<img width="968" alt="Image" src="https://github.com/user-attachments/assets/85b50cc0-673d-4cba-9902-f90589f88e37" />

**Comment by wenhuchen** [comment]
Sure, we could use another benchmark for validation in the revision. 

But just for simplicity, you can completely remove MATH results from our result tables. The improvements are still there. The most significant gains are in actually in Minerva-Math, OlympiadBench and AMC23. I am not sure whether this helps.

**Comment by LLMforScience** [comment]
But this is not the case. The improvement in MATH results is significant based on your table—ranking top-1  in DeepSeek-MATH and top-2 in Qwens.

<img width="1251" alt="Image" src="https://github.com/user-attachments/assets/8118b33a-8ca1-4907-9f48-6d7919fc705a" />

**Comment by wenhuchen** [comment]
In this table, both SFT and CFT adopt MATH-500 as the validation set. So it should be fair in some sense.
If you look at Qwen and Qwen-Math, the avg improvements are almost on par with the MATH improvement. I suspect that adopting another benchmark could lead to similar trend because the model is actually generalizing.

We could certainly add another table with other validation benchmarks in our revision. 

Going back to the motivation of using validation dataset, it's actually because of some characteristics of CFT. Training too long with CFT will lead the model to criticize everything without following instruction. The validation set was there simply to ensure that we didn't overtrain.

**Comment by LLMforScience** [comment]
Thanks for your quick responses.  I believe you should update the revision as soon as possible and include another table with results on different validation benchmarks to ensure a fair comparison.
